File: sortcharactersbyfrequency.py
class Solution(object):
    def frequencySort(self, s):
        """
        :type s: str
        :rtype: str
        """
        dic = {}
        for i in s:
            dic[i] = dic.get(i,0)+1
        pair = []
        for key in dic:
            pair.append((key, dic[key]))
        pair.sort(key=lambda x: x[1])
        res = ""
        while pair:
            s, n = pair.pop()
            res += s*n
        return res
        
        

File: cpp/sortcharactersbyfrequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        for (auto &c: s) count[c]++;
        sort(s.begin(), s.end(), [&](const char &a, const char &b){return count[a] > count[b] || (count[a] == count[b] && a < b);});
        return s;
    }
private:
    unordered_map<char, int> count;
};
